tests/test_model.py

```diff
@@ -21,29 +21,38 @@ def restore_default_dtype():
 @pytest.mark.parametrize("batch_size", (1, 3))
 @pytest.mark.parametrize("n_embd", (16, 32))
 @pytest.mark.parametrize("parallel_residual", (False, True))
-@pytest.mark.parametrize("kv_cache", (False, True))
 @pytest.mark.parametrize(
     ("device", "dtype"),
     [
         (torch.device("cpu"), torch.float32),
         pytest.param(
-            torch.device("cuda"),
-            torch.float16,
-            marks=[
-                pytest.mark.xfail(raises=AssertionError, strict=True),
-                pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA"),
-            ],
+            torch.device("cuda"), torch.float16, marks=[
+                # the reference does softmax upscaled to fp32 during attention. additionally, the final layernorm input
+                # is slightly different
```

Comment:
I wasn't able to find out why the final layernorm input is different. If you print `x.sum()` the value matches but some positions are not the same

Reply:
If I change the dtype for `cpu` test from `float32` to `blofat16` some combinations will fail.
<img width="740" alt="Screenshot 2023-10-24 at 7 24 22 PM" src="https://github.com/Lightning-AI/lit-gpt/assets/58434077/ab10335b-8191-40a6-ace3-d559356275b8">

Reply:
Probably for the same reason that the float16 tests are xfailed